frontend/packages/console-shared/src/components/formik-fields/EnvironmentField.tsx

```diff
@@ -47,6 +47,14 @@ const EnvironmentField: React.FC<EnvironmentFieldProps> = ({
     },
     [props.name, setFieldValue],
   );
+
+  React.useEffect(() => {
+    const reloadedEnvironmentVariables = !_.isEmpty(envs)
+      ? envs.map((env) => _.values(env))
+      : [['', '']];
+    values.formReloadCount && setNameValue(reloadedEnvironmentVariables);
```

Comment:
Put both lines inside an if statement, to not compute it

regarding formReloadCount always being true the useEffect is only called on change of envs or reload so is it alright ? if it is not, few ideas that i have are: 
- set reloadCount to zero inside the useEffect
- create a new state `lastReloadCount` and change if condition to `if (reloadCount != lastReloadCount)`
- eslint disable dependency and not include `envs` in my dependency

Reply:
I think we can exclude `envs` from the dependency.
and add a comment stating the reason that this effect only handles reload.
cc @rohitkrai03 

We could also wrap the logic for computing environment variables inside a function since it is being reused.

Reply:
thank you, made the changes